Tests/Common/Statistics/KellyCriterionManagerTests.cs

```diff
@@ -0,0 +1,125 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using NUnit.Framework;
+using QuantConnect.Statistics;
+
+namespace QuantConnect.Tests.Common.Statistics
+{
+    [TestFixture]
+    public class KellyCriterionManagerTests
+    {
+        [TestCase(1)]
+        [TestCase(0)]
+        public void ExtremeCasesBigNumbers(int extremeCase)
+        {
+            var extremePositive = new decimal[] { 1000, 2000, 3000, 1000, 5000, 1000, 1000 };
+            var extremeNegative = new decimal[] { -1000, -2000, -3000, -1000, -5000, -1000, -1000 };
+
+            decimal[] collection = extremeCase == 1 ? extremePositive : extremeNegative;
+
+            var kellyCriterionManager = new KellyCriterionManager();
+            foreach (var newValue in collection)
+            {
+                kellyCriterionManager.AddNewValue(newValue, DateTime.UtcNow);
+                kellyCriterionManager.UpdateScores();
+            }
+            var estimate = kellyCriterionManager.KellyCriterionEstimate;
+            var probabilityValue = kellyCriterionManager.KellyCriterionProbabilityValue;
+
+            Console.WriteLine($"Estimate {estimate} - ProbabilityValue {probabilityValue}");
+            Assert.AreEqual(extremeCase == 1
+                    ? 0.00031578947368421 : -0.00031578947368421, kellyCriterionManager.KellyCriterionEstimate);
+            Assert.AreEqual(1, kellyCriterionManager.KellyCriterionProbabilityValue);
+        }
+
+        [TestCase(1)]
+        [TestCase(0)]
+        public void ExtremeCasesSmallNumbers(int extremeCase)
+        {
+            var extremePositive = new decimal[] { 1, 2, 3, 1, 5, 1, 1 };
+            var extremeNegative = new decimal[] { -1, -2, -3, -1, -5, -1, -1 };
+
+            decimal[] collection = extremeCase == 1 ? extremePositive : extremeNegative;
+
+            var kellyCriterionManager = new KellyCriterionManager();
+            foreach (var newValue in collection)
+            {
+                kellyCriterionManager.AddNewValue(newValue, DateTime.UtcNow);
+                kellyCriterionManager.UpdateScores();
+            }
+
+            var estimate = kellyCriterionManager.KellyCriterionEstimate;
+            var probabilityValue = kellyCriterionManager.KellyCriterionProbabilityValue;
+
+            Console.WriteLine($"Estimate {estimate} - ProbabilityValue {probabilityValue}");
+            Assert.AreEqual(extremeCase == 1
+                ? 0.315789473684211m : -0.315789473684211m, kellyCriterionManager.KellyCriterionEstimate);
```

Comment:
I'm running the same calculations in Python and getting different numbers. For extreme large numbers I'm seeing an estimate of 0.000333333333 and with mean 2000.0 and variance 2000000. Same for small numbers case but on a different scale